### PR TITLE
Add helper functions for ipcidr

### PR DESCRIPTION
**Pull request by Issacwww**
**Description of changes:**
Add new helper function for EKS to config the template
have to reopen new PR due to #102 uses develop branch and need to discard original commit to sync with latest.

**Testing done:**
Added unit test and `cargo test` passed 

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: sources/api/schnauzer/Cargo.toml [hunk 1]
```diff
@@ -43,6 +43,7 @@ toml.workspace = true
 url.workspace = true
 bottlerocket-modeled-types.workspace = true
 settings-extension-oci-defaults.workspace = true
+cidr = "0.2.3"
```
Comment (cbgbt) [inline sources/api/schnauzer/Cargo.toml]: The CI should catch this when it runs, but you'll want to use workspace dependencies for this. Basically you'll want to:
* Add `cidr = 0.2` to the `[workspace.dependencies]` list in `sources/Cargo.toml`
* Change this to `cidr.workspace = true`.

`make twoliter check` should catch this too, if you want to try and replicate locally.

File: sources/api/schnauzer/src/helpers/mod.rs [hunk 2]
```diff
@@ -563,6 +570,123 @@ pub fn host(
...
+    let new_ip = Ipv4Addr::from(octets).to_string();
+    out.write(&new_ip).with_context(|_| error::TemplateWriteSnafu {
+        template: template_name.to_owned(),
+    })?;
+    
+    Ok(())
+}
+
+fn validate_and_parse_cidr(cidr: &str) -> Result<IpAddr, RenderError> {
+    let parsed_cidr: AnyIpCidr = cidr.parse().map_err(|_| error::InvalidCidrSnafu { cidr: cidr.to_string() }.build())?;
```
Comment (cbgbt) [inline sources/api/schnauzer/src/helpers/mod.rs]: If you mark `source` as [`NetworkParseError`](https://docs.rs/cidr/latest/cidr/errors/enum.NetworkParseError.html) in `InvalidCidrSnafu` you could just use `context` here, which would add additional information to the error message about why the parse failed.

**Review by cbgbt — COMMENTED** [review]
This looks great! There are only a few small changes I'd request.

Seems like the CI is failing due to unformatted sources. Mind running `cargo fmt` on `schnauzer`?

You can check run all of the CI lints, etc, ahead of time by running `make twoliter check`.

File: sources/api/schnauzer/Cargo.toml (hunk 1 above)
Comment (Issacwww) [reply sources/api/schnauzer/Cargo.toml]: Thanks for sharing this!
encountered in local run. Any clue what might be missed? I simply run the `make twoliter check` and testing looks good
```
[cargo-make] INFO - Running Task: check-migrations
grep: Release.toml: No such file or directory
[cargo-make] ERROR - Error while executing command, exit code: 2
[cargo-make] WARN - Build Failed.
Error: Command was unsuccessful, exit code 1
make: *** [twoliter] Error 1
```

Comment (cbgbt) [reply sources/api/schnauzer/Cargo.toml]: This actually seems to be a bug twoliter, sorry about that. I opened https://github.com/bottlerocket-os/twoliter/issues/359 to track it.

We'll provide a fix for that, but in the meantime you'll have to separately run the individual `check` tasks:  `make twoliter unit-tests check-fmt check-lints`l.

Comment (Issacwww) [reply sources/api/schnauzer/Cargo.toml]: Ah, the individual check looks good to me
`[cargo-make] INFO - Build Done in 52.05 seconds.`

**Review by cbgbt — APPROVED** [review]
Looks great! 🚀

**Review by yeazelm — APPROVED** [review]
LGTM! Thanks for the contribution!

**Pull request merged**